sbt-plugin/src/main/scala/org/scalajs/sbtplugin/ScalaJSPluginInternal.scala

```diff
@@ -655,26 +655,57 @@ private[sbtplugin] object ScalaJSPluginInternal {
       }
   )
 
+  private def isScala3(scalaV: String): Boolean =
+    scalaV.startsWith("3.")
+
   private val scalaJSProjectBaseSettings = Seq(
       platformDepsCrossVersion := ScalaJSCrossVersion.binary,
+      crossVersion := ScalaJSCrossVersion.binary,
 
       scalaJSModuleInitializers := Seq(),
       scalaJSUseMainModuleInitializer := false,
       jsEnvInput := Nil,
 
-      // you will need the Scala.js compiler plugin
-      addCompilerPlugin(
-          "org.scala-js" % "scalajs-compiler" % scalaJSVersion cross CrossVersion.full),
-
-      libraryDependencies ++= Seq(
-          // and of course the Scala.js library
-          "org.scala-js" %% "scalajs-library" % scalaJSVersion,
-          // as well as the test-bridge in the Test configuration
-          "org.scala-js" %% "scalajs-test-bridge" % scalaJSVersion % "test"
-      ),
+      /* Add core library dependencies (and compiler plugin), depending on the
+       * Scala version (2.x versus 3.x).
+       */
+      libraryDependencies := {
+        val prev = libraryDependencies.value
+        val scalaOrg = scalaOrganization.value
+        val scalaV = scalaVersion.value
+
+        if (isScala3(scalaV)) {
+          /* Remove scala3-library (non _sjs1) in case sbt-dotty was applied
+           * before sbt-scalajs.
+           */
+          val filteredPrev = prev.filterNot { dep =>
+            dep.organization == scalaOrg && dep.name == "scala3-library"
+          }
+          filteredPrev ++ Seq(
+              scalaOrg %% "scala3-library_sjs1" % scalaV,
+              "org.scala-js" % "scalajs-library_2.13" % scalaJSVersion, // bump to at least scalaJSVersion
```

Comment:
I do not understand the comment here. As in: why does this line need that comment and none of the others?

Reply:
I've elaborated the comment. Is it clearer now?